src/python/pants/engine/target.py

```diff
@@ -410,6 +412,7 @@ def __init__(
             like `dir:` know whether to match the target or not.
         :param ignore_unrecognized_fields: Don't error if fields are not recognized. This is only
             intended for when Pants is bootstrapping itself.
+        :param build_file_source: Path to BUILD file and line number where this target is declared.
```

Comment:
@stuhood are you saying you want to rename this parameter?

Would you prefer `build_file_origin`?

```suggestion
        :param build_file_origin: Path to BUILD file and line number where this target is declared.
```